detection/src/main/java/io/herow/sdk/detection/notification/NotificationHelper.kt

```diff
@@ -0,0 +1,175 @@
+        }
+
+        GlobalLogger.shared.debug(null, "Result sent is: $result")
+
+        return result
+    }
+
+    private fun dynamicValues(sentenceToAnalyze: String): DynamicResult {
+        GlobalLogger.shared.debug(null, "Regex is: $sentenceToAnalyze")
+        var couple = HashMap<String, String>()
+
+        val newText = sentenceToAnalyze
+            .replace("{{", "")
+            .replace("}}", "")
+
+        GlobalLogger.shared.debug(null, "Result is: $newText")
+
+        val components = newText.split("|")
+        GlobalLogger.shared.debug(null, "Components to return: $components")
+
+        val sentenceNotExtracted = sentenceToAnalyze
+            .replace("default('", "")
+            .replace("')", "")
+
+        if (components.size > 1) {
+            couple = extractDefaultValue(sentenceNotExtracted)
+        }
+
+        return DynamicResult(removeRegex(newText), couple)
+    }
+
+    private fun removeRegex(sentence: String): String {
+        val regex = "\\|([a-z A-Z0-9]*)\\('[a-z A-Z0-9-]*'\\)"
+        val pattern = Pattern.compile(regex, Pattern.MULTILINE)
+        val matcher = pattern.matcher(sentence)
+
+        return matcher.replaceAll("")
+    }
+
+    private fun extractDefaultValue(sentence: String): HashMap<String, String> {
+        val couple = HashMap<String, String>()
+        var searchingPosition = 0
+
+        while (searchingPosition >= 0) {
+            var insertingKey: String
+            var insertingValue: String
+
+            searchingPosition = sentence.indexOf("{{", searchingPosition)
+
+            if (searchingPosition > -1) {
+                val separator = sentence.indexOf("|", searchingPosition)
+                insertingKey = sentence.substring(searchingPosition + 2, separator)
+
+                GlobalLogger.shared.debug(null, "InsertingKey is: $insertingKey")
+
+                val endPosition = sentence.indexOf("}}", searchingPosition)
+                insertingValue = sentence.substring(separator + 1, endPosition)
+
+                GlobalLogger.shared.debug(null, "InsertingValue is: $insertingValue")
+                couple[insertingKey] = insertingValue
+                searchingPosition = endPosition + 2
+            }
+        }
+
+        GlobalLogger.shared.debug(null, "Couple is: $couple")
+
+        return couple
+    }
+
+    fun canCreateNotification(campaign: Campaign, sessionHolder: SessionHolder): Boolean {
+        if (!TimeSlotFilter.createNotification(
+                campaign,
+                sessionHolder
+            ) || !(DayRecurrencyFilter.createNotification(
+                campaign, sessionHolder
+            )) || !(ValidityFilter.createNotification(
+                campaign,
+                sessionHolder
+            )) || !(CappingFilter.createNotification(campaign, sessionHolder))
+        ) {
+            return false
+        }
+
+        GlobalLogger.shared.warning(null, "Can create notification")
+        return true
+    }
+}
+
```

Comment:
dans cette methode tu lui dis directement de passer dans les filtres existants  j aurais preféré que le notificationManager ai une liste de filtres ( un tableau) dans lequel  tu ajoutes tes filtres  avec un registerFilter.( pour plus tard en recreer un autre au besoin tu n auras que le register et voilà sans prise de tête.
Que ce soit directement le notificationManager qui itère sur ses filtres  et pas ton objet notificationHelper qui lui fait  :    if (!TimeSlotFilter.createNotification(
                campaign,
                sessionHolder
            ) || !(DayRecurrencyFilter.createNotification(
                campaign, sessionHolder
            )) || !(ValidityFilter.createNotification(
                campaign,
                sessionHolder
            )) || !(CappingFilter.createNotification(campaign, sessionHolder))
        ) {
            return false
        }

tu itères sur tous les filtres et dès que y en a un qui te répond false tu retournes le résultat.


un truc du genre: 

   private func canCreateNotification( _ campaign : Campaign) -> Bool {
        for filter in filters {
                if !(filter.createNotification(campaign: campaign, completion: nil)) {
                    return false
                }
        }
        GlobalLogger.shared.warning("can reate notification")
        return true
    }